### PR TITLE
[WNMGDS-531] Add autofocus and ref to <Alert>

### PR DESCRIPTION
**Pull request by nichia**
### Add
- add `alertRef` prop to <Alert> div block
- add `autoFocus` prop to autofocus on the <Alert> div block on initial loading of the alert
- support other undefined props to the Alert component
 
The props perform similar function as 'inputRef' & 'focusTrigger' on other components

## How to test
1. Run the doc site locally 
1. Add the `alertRef={(elem) => console.log(elem)}` prop to the `<Alert>` example 
1. Add the `autoFocus` prop to the `<Alert>` example 
1. See that alert code block is printed on console and focus is activated on mount of Alert page

Demo website: http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-531/Add-autofocus-and-ref-to-alert/

## NOTE
Checking with Josh Nelson to determine if there's a case for `alertRef` prop or would the `autoFocus` to focus on first mount be sufficient?

### CODE REVIEW

File: packages/design-system/src/components/Alert/Alert.jsx [hunk 1]
```diff
@@ -40,6 +47,18 @@ export class Alert extends React.PureComponent {
...
+          if (this.props.focusTrigger) {
+            this.focusRef = ref;
+          } else {
+            if (this.props.inputRef) {
+              this.props.inputRef(ref);
+            }
+          }
+        }}
+        /* eslint-enable no-return-assign */
+        tabIndex={this.props.inputRef || this.props.focusTrigger ? '-1' : null}
```
[the remark this answers is not in the document]
Comment (bernardwang) [reply packages/design-system/src/components/Alert/Alert.jsx:61]: I think this is ok considering it only happens when the ref or focus prop is used

**Review by line47 — APPROVED** [review]
This looks good to me!

**Comment by bernardwang** [comment]
@nichia please update this per our convo and when the reporter responds. I don't believe this is ready to review yet

**Comment by nichia** [comment]
PR closed - inactivity from reporter.

**Review by nichia — COMMENTED** [review]
@pwolfert - do you think this PR as is, addresses the `I wonder if a focusOnRender or focus prop on CMSDS alerts would be helpful generally` discussed in this slack? https://cmsgov.slack.com/archives/CHH0381RD/p1618872677058400

**Comment by pwolfert** [comment]
@nichia, this is great! Yeah, it would solve the problem we have. I'm glad your code isn't going to be thrown away!

File: packages/design-system/src/components/Alert/Alert.jsx [hunk 2]
```diff
@@ -30,22 +37,49 @@ export class Alert extends React.PureComponent {
...
       <div
         className={classes}
-        role={this.props.role}
-        aria-labelledby={this.props.heading ? this.headingId : undefined}
+        /* eslint-disable no-return-assign */
+        ref={(ref) => {
+          if (autoFocus) {
+            this.focusRef = ref;
+          } else {
+            if (alertRef) {
```
Comment (pwolfert) [inline packages/design-system/src/components/Alert/Alert.jsx:69]: Can this be combined with the `else` line above it?

**Review by pwolfert — APPROVED** [review]
I just tested out both props, and they're working for me 👍  (I didn't test it in the context of one of the apps, but I'm confident it'll meet our needs.)

**Comment by nichia** [comment]
@davidakennedy - do you see any other accessibility changes needed in making an an alert focusable?

**Review by davidakennedy — APPROVED** [review]
@nichia I can't think of any other accessibility implications here. I added a commit to add a note to the docs regarding shifting focus. Let me know if that doesn't make sense.

**Pull request merged**